web/src/message_fetch.ts

```diff
@@ -347,14 +347,18 @@ export function load_messages(opts: MessageFetchOptions, attempt = 1): void {
         for (const term of terms) {
             if (term.operator === "channel") {
                 const sub = stream_data.get_sub_by_id_string(term.operand);
+                if (!sub) {
+                    narrow_banner.show_empty_narrow_message();
+                    return;
```

Comment:
There are other invalid narrows that we might not want to fetch messages for from the server (two terms with "channel" operators for example), so would it make sense to do a check for an invalid narrow before calling `load_messages`?

Also, with the changes in #31934 to address the `TODO` of sending channel IDs to the server, this block would be removed from this function as we expect the term with the "channel" operator to have a valid string ID or be an empty string.

Reply:
Ah right, so I guess I can just remove this commit? Adding something for validating the narrow sounds like a good idea, maybe as a separate PR?

Reply:
#31934 is merged, so you can rebase on top of that.

Reply:
I can see those changes being a separate PR and this one just updating the text for these empty narrows.